hibernate-core/src/test/java/org/hibernate/orm/test/loading/multiLoad/MultiLoadSubSelectCollectionTest.java

```diff
@@ -88,25 +88,11 @@ public void testSubselect(SessionFactoryScope scope) {
 						assertFalse( Hibernate.isInitialized( p.children ) );
 					}
 
-					// When the first collection is loaded, the full batch of 50 collections
+					// When the first collection is loaded, the full collection
 					// should be loaded.
 					Hibernate.initialize( list.get( 0 ).children );
 
-					for ( int i = 0; i < 50; i++ ) {
-						assertTrue( Hibernate.isInitialized( list.get( i ).children ) );
-						assertEquals( i + 1, list.get( i ).children.size() );
-					}
-
-					// The collections for the 51st through 56th entities should still be uninitialized
-					for ( int i = 50; i < 56; i++ ) {
-						assertFalse( Hibernate.isInitialized( list.get( i ).children ) );
-					}
-
-					// When the 51st collection gets initialized, the remaining collections should
-					// also be initialized.
-					Hibernate.initialize( list.get( 50 ).children );
-
-					for ( int i = 50; i < 56; i++ ) {
+					for ( int i = 0; i < 56; i++ ) {
```

Comment:
How about using a special dialect here that has a limit of 50 to retain the same test logic?

Reply:
I think we have to change `BatchLoadSizingStrategy` implementation, at the moment we are not taking into account any Dialect limit

Reply:
It is taken into account, here: https://github.com/hibernate/hibernate-orm/pull/4310/files#diff-db73bbf122356d881edeb9b3d69cc9ad5b469229214d289131fd6c98c6b7b4c0R3077

Reply:
Done!